### PR TITLE
Set up MongoDB and basic models

### PR DESCRIPTION
**Pull request by ashleywd**

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        type: Number, 
+        default: 0 
+    },
+    createdAt: { 
+        type: Date, 
+        default: Date.now 
+    }
+});
+
+module.exports = mongoose.model('User', userSchema);
```
Comment (daosgava) [inline models/user.js:28]: This model looks good :)

File: models/post.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    acceptedBy: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'User',
+        default: null
+    }
+}, {
+    timestamps: true
+});
+
+module.exports = mongoose.model('Post', postSchema);
```
Comment (daosgava) [inline models/post.js:36]: I think we will need a property to specify the number of reward points. By the way, does the user need points to request a job?

File: server.js [hunk 3]
```diff
@@ -3,21 +3,12 @@ const express = require('express');
 const mongoose = require('mongoose');
 const path = require('path');
 const session = require('express-session');
+const User = require('/models/user');
+const Post = require('/models/post');
```
Comment (daosgava) [inline server.js:7]: I think these models need to be used in the controller to follow the MVC architecture pattern, not in the server.js.

File: server.js [hunk 4]
```diff
@@ -30,26 +21,72 @@ app.use(session({
...
-  .catch(err => console.error('MongoDB connection error:', err));
+const mongoURI = process.env.MONGODB_URI || 'mongodb://localhost:27017/babyswap';
+mongoose.connect(mongoURI).then(() => {
+    console.log('Connected to MongoDB successfully');
+    // Optional: Create test data
+    createTestData();
+}).catch(err => {
+    console.error('MongoDB connection error:', err);
+    process.exit(1);  // Exit if cannot connect to database
+});
```
Comment (daosgava) [inline server.js:32]: Same I think we should move this to a utils file. Take a look to the bootstrap project provided by the lecturer. https://github.com/alamfaisal654/SIT725/blob/main/Week-5/dbConnection.js

File: server.js [hunk 5]
```diff
@@ -30,26 +21,72 @@ app.use(session({
...
+    res.sendFile(path.join(__dirname, 'public', 'posts.html'));
+});
+
+app.get('/login', (req, res) => {
+    res.sendFile(path.join(__dirname, 'public', 'login.html'));
+});
+
+app.get('/register', (req, res) => {
+    res.sendFile(path.join(__dirname, 'public', 'register.html'));
+});
```
Comment (daosgava) [inline server.js:79]: I think we need to include this in a router file. Check this code https://github.com/alamfaisal654/SIT725/blob/main/Week-5/routers/router.js

Something like 

routers/app.js

```
let express = require('express');
let router = express.Router();

router.post('/register',(req,res) =>{
...
});

router.get('/posts', (req,res)=>{
...
});

router.get('/login', (req,res)=>{
...
});

```

File: server.js [hunk 6]
```diff
@@ -30,26 +21,72 @@ app.use(session({
...
 app.use((err, req, res, next) => {
     console.error(err.stack);
-    res.status(500).json({ error: '500 Server Error' });
+    res.status(500).json({ error: 'Something went wrong!' });
 });
 
 // 404 handler
 app.use((req, res) => {
-    res.status(404).json({ error: '404 Not Found' });
+    res.status(404).json({ error: 'Route not found' });
```
Comment (daosgava) [inline server.js:89]: Nice!

**Review by daosgava — CHANGES_REQUESTED** [review]
Hey @ashleywd great job! I’ve added a few comments to address the MVC pattern requirements outlined in the project group rubric.

File: models/user.js (hunk 1 above)
Comment (ashleywd) [reply models/user.js:28]: Great

File: models/post.js (hunk 2 above)
Comment (ashleywd) [reply models/post.js:36]: Ideally yes, however with no starting points no user will be able to post. So we may need to say that any starting user recieves points by default.

File: server.js (hunk 3 above)
Comment (ashleywd) [reply server.js]: Yes I think these were only here for testing. I will move these to the controller.

Comment (ashleywd) [reply server.js]: (Confirmed moved to a controller.js file now). Will commit the changes shortly.

File: server.js (hunk 4 above)
Comment (ashleywd) [reply server.js]: Created a db.js file under /config/

**Pull request closed without merging**